neo4j-jdbc-bolt/src/test/java/org/neo4j/jdbc/bolt/cache/BoltDriverCacheTest.java

```diff
@@ -20,6 +33,43 @@
 
     Function<BoltDriverCacheKey, Driver> builder = (params -> Mockito.mock(Driver.class));
 
+    @Test
+    public void cacheShouldBeThreadSafe() throws ExecutionException, InterruptedException {
+        List<URI> url = Arrays.asList(URI.create("bolt://localhost"));
+        Config.ConfigBuilder configBuilder = Config.builder();
+        AuthToken authToken = AuthTokens.basic("neo4j", "password");
+
+        BoltDriverCache cache = new BoltDriverCache(builder);
+        assertEquals(0, cache.getCache().size());
+
+        int numThreads = Runtime.getRuntime().availableProcessors();
+        ExecutorService executor = Executors.newWorkStealingPool();
+
+        AtomicBoolean running = new AtomicBoolean();
+        AtomicInteger overlaps = new AtomicInteger();
+
+        Collection<Callable<Driver>> getDriver = new ArrayList<>();
+        for (int t = 0; t < numThreads; ++t) {
+            getDriver.add(() -> {
+                if (!running.compareAndSet(false, true)) {
+                    overlaps.incrementAndGet();
+                }
+                Driver d = cache.getDriver(url, configBuilder.build(), authToken, new Properties());
+                running.compareAndSet(true, false);
+                return d;
+            });
+        }
+
+        Map<Driver, Integer> drivers = new IdentityHashMap<>();
+        for (Future<Driver> getDriverFuture : executor.invokeAll(getDriver)) {
+            drivers.put(getDriverFuture.get(), 1);
+        }
+
+        assertTrue(overlaps.get() > 0); // This might get flaky...
```

Comment:
it is indeed flaky
last CI build at the time of writing failed with this assertion
any idea on how to make it more stable? create more callables?

Reply:
@michael-simons the GH action container has just 2 cores, maybe (as @fbiville suggested) makes sense just to double (or triple) the callables, wdyt?

Btw can you also rebase? we fixed a flaky test related to an old PowerMock version that can affect the CI if your test become green